### ISSUE TITLE
Cannot save changes on Firefox

### ISSUE THREAD

**Issue by lewisxy** [observablehq/feedback#316]
**Describe the bug**
Observable cannot save changes on Firefox. I have tested it with Firefox 95.0 on Linux and Firefox 95.0.1 on MacOS. It works fine on Safari though.

When I open a notebook, the circle next to "publish" button is keep rotating, and an error message indicating it cannot save change will pop up when I try to modify the notebook.

When I open JavaScript console from the web developer tool of Firefox, this error message keep popping up.
```
Firefox can’t establish a connection to the server at wss://ws.observablehq.com/document/<...>/edit.
```

**To Reproduce**
1. Create a notebook (or open a notebook) in the Firefox of specified version above.
2. Observe the behavior and error message in developer console

**Additional context**
I should note that I do have a SOCKS proxy and configured Firefox to use system proxy (by default). However, the Safari does not have this problem when using the same proxy and Firefox has no problem opening any other websites that uses websocket.

**Comment by CobusT** [comment]
Does it work without the SOCKS proxy? I assume so...

**Comment by lewisxy** [comment]
> Does it work without the SOCKS proxy? I assume so...

It does appear to work.

But I remember it works a few weeks ago (on Firefox behind my proxy) before I did the update to 95. I am not sure if it's a Firefox issue in its latest update if the Observablehq side haven't change anything.

**Comment by lewisxy** [comment]
After some trials and errors, I found out the problem.

TLDR: SOCKS proxy doesn't play well with IPv6

When I opened `about:network` in Firefox, under `DNS` tab on the right, it showed all the DNS resolution record. Apparently, Firefox uses DoH (DNS-over-HTTPS) and it will prefer IPv6 address if possible. This is what happen in my case. `ws.observablehq.com` get resolved into IPv6 addresses, and apparently my SOCKS proxy can't handle that. 

I am still not sure what's the best solution in this case. A temporary solution will be to disable IPv6 support in Firefox. In `about:config`, change `network.dns.disableIPv6` to `true`. After this, DNS tab in `about:network` showed that `ws.observablehq.com` has been resolved to IPv4 address and it works fine. However, in the long term, it would be nice if my SOCKS proxy can handle IPv6 as well. I am not familiar enough with SOCKS proxy to configure it yet, and I will work on that later. In any case. Thanks for the support.
